### PR TITLE
DM/Openapi: improve password input mechanism source

### PR DESCRIPTION
**Pull request by WizardXiao**
### What problem does this PR solve?

Issue Number: ref #4986

### What is changed and how it works?
source's password can be nil. 
1. password will be set "" when create source with password which is nil.
2. password will use old password when update source with password which is nil.

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: dm/openapi/spec/dm.yaml [hunk 1]
```diff
@@ -1376,7 +1376,7 @@ components:
           type: string
           example: "123456"
           description: "source password"
-          nullable: false
+          nullable: true
```
Comment (Ehco1996) [inline dm/openapi/spec/dm.yaml:1379]: also need to remove reqired filed `password` for source, 
it seems that openapi will return 400 when input  json don't have this filed, maybe add a it-test for this

**Review by Ehco1996 — COMMENTED** [review]
restLGTM

File: dm/dm/master/openapi_controller.go [hunk 2]
```diff
@@ -135,6 +135,12 @@ func (s *Server) updateSource(ctx context.Context, sourceName string, req openap
 		return nil, terror.ErrSchedulerSourceCfgNotExist.Generate(sourceName)
 	}
 	newCfg := config.OpenAPISourceToSourceCfg(req.Source)
+
+	// req has no password, use old password
```
[the remark this answers is not in the document]
Comment (WizardXiao) [reply dm/dm/master/openapi_controller.go:139]: ok, this is for `updateSource` and use don't want to input password again. here use nil pointer to make a distinction between `no input` and `inupu is ""`.

File: dm/dm/config/source_converter.go [hunk 3]
```diff
@@ -23,7 +24,7 @@ func SourceCfgToOpenAPISource(cfg *SourceConfig) openapi.Source {
 		Enable:     cfg.Enable,
 		EnableGtid: cfg.EnableGTID,
 		Host:       cfg.From.Host,
-		Password:   "******", // PM's requirement, we always return obfuscated password to user
+		Password:   utils.StringP("******"), // PM's requirement, we always return obfuscated password to users
```
[the remark this answers is not in the document]
Comment (WizardXiao) [reply dm/dm/config/source_converter.go:27]: yes, that will be better. i will fix.

File: dm/openapi/spec/dm.yaml (hunk 1 above)
Comment (WizardXiao) [reply dm/openapi/spec/dm.yaml:1379]: yes. i forget. i will fix and add a it-test.

File: dm/dm/master/openapi_controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (D3Hunter) [reply dm/dm/master/openapi_controller.go:139]: i mean in the comment

[the remark this answers is not in the document]
Comment (WizardXiao) [reply dm/dm/master/openapi_controller.go:139]: ok, i will add to comment.

File: dm/dm/config/source_config.go [hunk 4]
```diff
@@ -53,6 +53,10 @@ var getAllServerIDFunc = utils.GetAllServerID
 //go:embed source.yaml
 var SampleSourceConfig string
 
+// DefaultFeedbackEnPassword is the source encryption password that returns to the foreground.
+// PM's requirement, we always return obfuscated password to users.
+var DefaultFeedbackEnPassword string = "******"
```
Comment (D3Hunter) [inline dm/dm/config/source_config.go:58]: maybe ObfuscatedPasswordForFeedback?

Comment (WizardXiao) [reply dm/dm/config/source_config.go:58]: yes, more appropriate.

**Review by D3Hunter — APPROVED** [review]

**Review by Ehco1996 — APPROVED** [review]
LGTM

**Pull request merged**